packages/contracts/source/contracts/serializer.ts

```diff
@@ -1,19 +1,18 @@
 import { ByteBuffer } from "@mainsail/utils";
 
-export type PropertyType =
-	| "uint32"
-	| "uint64"
-	| "address"
-	| "bigint"
-	| "hash"
-	| "publicKey"
-	| "signature"
-	| "transactions";
-
-export interface SerializationSchema {
-	type: PropertyType;
-	required?: boolean;
-}
+export type MandatoryPropertyType = "uint32" | "uint64" | "address" | "bigint" | "hash" | "publicKey" | "hex";
+
+export type OptionalPropertyType = "blockId" | "signature" | "transactions";
+
+export type SerializationSchema =
+	| {
+			type: MandatoryPropertyType;
+			required: true;
```

Comment:
Maybe `optional` field can be used instead required. Field is required by default, except for the blockId. In that case schema should specify how blockId its handled via mandatory optional field.

Reply:
Another option is to introduce new field like: `optionalBlockId` and drop `required` field.